core/src/main/scala/org/apache/spark/deploy/security/HadoopFSDelegationTokenProvider.scala

```diff
@@ -150,6 +150,7 @@ private[deploy] class HadoopFSDelegationTokenProvider
         val interval = newExpiration - getIssueDate(tokenKind, identifier)
         logInfo(log"Renewal interval is ${MDC(TOTAL_TIME, interval)} for" +
           log" token ${MDC(TOKEN_KIND, tokenKind)}")
+        token.cancel(hadoopConf)
```

Comment:
have you tested it? it requires some time to propagate the new token from driver to executors, I think such a change may cause auth failures on the executor side

Reply:
Thanks for your review! These tokens have not actually been used. The tokens that are truly distributed to each executor are obtained as Line57:
https://github.com/apache/spark/blob/dd259b0b27841e6dd7c07f8ca3cc05d275863dd5/core/src/main/scala/org/apache/spark/deploy/security/HadoopFSDelegationTokenProvider.scala#L57-L64

Reply:
ha, that makes sense